### PR TITLE
Extract reusable components from Stage0 DICE code

### PR DESCRIPTION
**Pull request by conradgrobler**
Many parts of the existing Stage 0 DICE certificate generation code can be reused for other boot stages. This change refactors the code so that the reusable parts can be moved to a shared crate (oak_dice).

### CODE REVIEW

File: oak_dice/src/cert.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    let ikm = public_key.to_encoded_point(false);
+    let hkdf = Hkdf::<Sha256>::new(Some(ID_SALT), ikm.as_bytes());
+    let mut result = [0u8; KEY_ID_LENGTH];
+    hkdf.expand(INFO_STR, &mut result)
+        .expect("invalid length for HKDF output");
+    result
+}
+
+/// Generates private/public ECDSA key pair.
+pub fn generate_ecdsa_keys() -> (SigningKey, VerifyingKey) {
```
Comment (ipetr0v) [inline oak_dice/src/cert.rs:87]:
```suggestion
pub fn generate_ecdsa_key_pair() -> (SigningKey, VerifyingKey) {
```

Comment (ipetr0v) [reply oak_dice/src/cert.rs:87]: nit

**Review by ipetr0v — APPROVED** [review]

File: oak_dice/src/cert.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+//! Constants and helper functions to work with CWT-based DICE certificates.
+
+use alloc::{string::String, vec, vec::Vec};
+use coset::{
+    cbor::value::Value,
+    cwt::{ClaimName, ClaimsSetBuilder},
+    iana, Algorithm, CborSerializable, CoseError, CoseKey, CoseSign1, KeyOperation, KeyType, Label,
+};
+use hkdf::Hkdf;
+use p384::ecdsa::{signature::Signer, Signature, SigningKey, VerifyingKey};
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply oak_dice/src/cert.rs]: Switched to p256.

File: oak_dice/src/cert.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_dice/src/cert.rs]: Done

**Review by conradgrobler — COMMENTED** [review]
Thanks

**Pull request merged**